### PR TITLE
Make interpreter async

### PR DESCRIPTION
**Pull request by mikevoronov**
Resolves https://github.com/fluencelabs/aquavm/issues/140.

### CODE REVIEW

File: avm/server/src/lib.rs [hunk 1]
```diff
@@ -46,7 +45,13 @@ pub use fluence_faas::HostImportError;
 pub use fluence_faas::IType;
 pub use fluence_faas::IValue;
 
+pub use air_interpreter_interface::CallRequests;
+pub use air_interpreter_interface::CallResults;
 pub use air_interpreter_interface::InterpreterOutcome;
```
Comment (folex) [inline avm/server/src/lib.rs:50]: Is there any reason to keep InterpreterOutcome in exports of avm-server? I'd think AVMOutcome supersedes it

File: crates/interpreter-interface/src/call_request_parameters.rs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+pub type CallRequests = HashMap<u32, CallRequestParams>;
+
+#[marine]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct CallRequestParams {
+    pub service_id: String,
+    pub function_name: String,
+    pub arguments: String,
+    pub tetraplets: String,
```
Comment (folex) [inline crates/interpreter-interface/src/call_request_parameters.rs:30]: Let's add some docs here pls

File: avm/server/src/lib.rs (hunk 1 above)
Comment (mikevoronov) [reply avm/server/src/lib.rs:50]: Nope, you're right.

File: crates/interpreter-interface/src/call_request_parameters.rs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+#[marine]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct CallRequestParams {
+    /// Id of a service that should be called.
+    pub service_id: String,
+
+    /// Name of a function from service identified by service_id that should be called.
+    pub function_name: String,
+
+    /// Serialized to JSON string Vec<IValue> of arguments that should be passed to a service.
```
Comment (folex) [inline crates/interpreter-interface/src/call_request_parameters.rs:35]: It's a serialized Vec<JValue>, right? `IValue` is a typo

Comment (mikevoronov) [reply crates/interpreter-interface/src/call_request_parameters.rs:35]: Yes, you are right, thanks!

**Review by coder11 — APPROVED** [review]

**Pull request merged**